test/unit/vl-progress-bar.test.html

```diff
@@ -0,0 +1,123 @@
+<!doctype html>
+<html>
+
+<head>
+  <meta charset="utf-8">
+  <script src="../../../@webcomponents/webcomponentsjs/webcomponents-lite.js"></script>
+  <script src="../../../web-component-tester/browser.js"></script>
+
+  <script type="module" src="../../src/vl-progress-bar.js"></script>
+  <script type="module" src="../../src/vl-progress-bar-item.js"></script>
+</head>
+
+<body>
+  <test-fixture id="vl-progress-bar-fixture">
+    <template>
+      <vl-progress-bar>
+        <vl-progress-bar-item>Stap 1/2: Aanvraag</vl-progress-bar-item>
+        <vl-progress-bar-item>Stap 2/2: Bevestigen</vl-progress-bar-item>
+      </vl-progress-bar>
+    </template>
+  </test-fixture>
+
+  <script>
+    suite('<vl-progress-bar>', () => {
+      const should = chai.should();
+
+      test('voor elke progress-bar-item zal een progress bar stap gegenereerd worden', () => {
+        const progressBar = fixture('vl-progress-bar-fixture');
+        const progressBarItems = progressBar.querySelectorAll('vl-progress-bar-item');
+        progressBarItems.forEach((item, index) => {
+          const step = progressBar.shadowRoot.querySelectorAll('.vl-progress-bar__step')[index];
+          const bullet = step.querySelector('.vl-progress-bar__bullet');
+          should.exist(step);
+          should.exist(bullet);
+          const containsActiveClass = step.classList.contains('vl-progress-bar__step--active');
+          index == 0 ? assert.isTrue(containsActiveClass) : assert.isFalse(containsActiveClass);
+        });
+      });
+
+      test('voor elke progress-bar-item zal een tooltip gegenereerd worden', () => {
+        const progressBar = fixture('vl-progress-bar-fixture');
+        const progressBarItems = progressBar.querySelectorAll('vl-progress-bar-item');
+        progressBarItems.forEach((item, index) => {
+          const step = progressBar.shadowRoot.querySelectorAll('.vl-progress-bar__step')[index];
+          const tooltip = step.querySelector('vl-tooltip');
+          should.exist(tooltip);
+          assert.equal(tooltip.getAttribute('placement'), 'top');
+          assert.equal(tooltip.textContent, item.text);
+        });
+      });
```

Comment:
dit lijken mij eerder testen te moeten zijn van de step?

Reply:
Dat is dubbel, het is een samenspel van vl-ui-progress-bar en vl-ui-progress-bar-item. Ik zou de template in isolatie kunnen unit testen op vl-ui-progress-bar-item niveau, maar nu test ik dit in de vl-ui-progress-bar waardoor ik meteen ook test dat ze goed aangeroepen worden. Opsplitsen zal dan weer wat duplicatie tot gevolg hebben? Wat denk je?

Reply:
voor mij test ne unit test ne unit